ReceiptRepository/Tests/PrepaidCardReceiptRepositoryTests.swift

```diff
@@ -0,0 +1,88 @@
+// sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Hippolyte
+import HyperwalletSDK
+@testable import ReceiptRepository
+import XCTest
+
+class PrepaidCardReceiptRepositoryTests: XCTestCase {
+    private lazy var listPrepaidCardReceiptPayload = HyperwalletTestHelper.getDataFromJson("PrepaidCardReceiptResponse")
+    private var factory: ReceiptRepositoryFactory!
+    private var prepaidCardReceiptRepository: PrepaidCardReceiptRepository!
+    private var receiptExpectation: XCTestExpectation!
+
+    override func setUp() {
+        Hyperwallet.setup(HyperwalletTestHelper.authenticationProvider)
+        factory = ReceiptRepositoryFactory.shared
+        prepaidCardReceiptRepository = factory.prepaidCardReceiptRepository()
+        receiptExpectation = self.expectation(description: "load prepaid card receipts")
+    }
+
+    override func tearDown() {
+        ReceiptRepositoryFactory.clearInstance()
+        if Hippolyte.shared.isStarted {
+            Hippolyte.shared.stop()
+        }
+    }
+
+    func testListUserReceiptSuccess() {
+        var prepaidCardReceiptList = [HyperwalletReceipt]()
+        HyperwalletTestHelper.setUpMockServer(
+            request: ReceiptRequestHelper.setUpRequest(listPrepaidCardReceiptPayload,
+                                                       nil,
+                                                       "trm-123456789"))
+        prepaidCardReceiptRepository.listPrepaidCardReceipts(
+            prepaidCardToken: "trm-123456789"
+        ) { [weak receiptExpectation ] result in
+            switch result {
+            case .success(let receiptPageList):
+                guard let receiptPageList = receiptPageList else {
+                    XCTFail("The User's receipt list should not be empty")
+                    return
+                }
+                receiptExpectation?.fulfill()
+                prepaidCardReceiptList = receiptPageList.data
+
+            case .failure:
+                XCTFail("Unexpected error")
+            }
+        }
+        wait(for: [receiptExpectation], timeout: 1)
+        XCTAssertFalse(prepaidCardReceiptList.isEmpty, "The Prepaid Card receipt list should not be empty")
+    }
+
+    func testListUserReceiptWhenErrorReceived() {
```

Comment:
rename to testListPrepaidCardReceipt_failure

Reply:
renamed